### PR TITLE
Fixed credit mining bug about non-existing path

### LINKED ISSUE #3837: WindowsError: [Error 3] The system cannot find the path specified
From reporter - Tribler 7.1.0-beta [#15897](http://reporter.tribler.org/stacktrace/15897)
Windows-10-10.0.17134-SP0 - 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)]
```python
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 127, in on_read_data
RuntimeError: [Failure instance: Traceback: <type 'exceptions.WindowsError'>: [Error 3] The system cannot find the path specified
lib\site-packages\twisted\internet\base.py:1243:run
lib\site-packages\twisted\internet\base.py:1252:mainLoop
lib\site-packages\twisted\internet\base.py:878:runUntilCurrent
lib\site-packages\twisted\internet\task.py:239:__call__
--- <exception caught here> ---
lib\site-packages\twisted\internet\defer.py:150:maybeDeferred
Tribler\Core\CreditMining\CreditMiningManager.py:98:check_disk_space
Tribler\Core\CreditMining\CreditMiningManager.py:93:get_free_disk_space
lib\site-packages\psutil\__init__.py:2040:disk_usage
lib\site-packages\psutil\_pswindows.py:249:disk_usage
]
```

### PR DESCRIPTION
**Pull request by xoriole**
Fixed credit mining bug about the non-existing path.
Fixes #3837

### CODE REVIEW

File: Tribler/Core/CreditMining/CreditMiningManager.py [hunk 1]
```diff
@@ -92,7 +92,30 @@ def shutdown(self, remove_downloads=False):
...
+        # Check that credit mining directory exists, if not try to re-create it.
+        if not os.path.exists(self.settings.save_path):
+            try:
+                os.makedirs(self.settings.save_path)
+                error_message = "Credit mining directory [%s]  does not exist. Tribler will re-create " \
+                                "the directory and resume again.<br/>If you wish to disable credit mining entirely, " \
+                                "please go to Settings >> ANONYMITY >> Token mining. " % self.settings.save_path
+            except OSError:
+                self.shutdown()
+                error_message = "Credit mining directory [%s] was deleted or does not exist and Tribler could not " \
```
Comment (qstokkink) [inline Tribler/Core/CreditMining/CreditMiningManager.py]: Does this work with unicode paths?

**Comment by xoriole** [comment]
retest this please

**Comment by xoriole** [comment]
retest this please

**Comment by devos50** [comment]
Hmm, linux VNC server seems to fail in GUI tests. I've seen this before but it's very rare. Maybe we should check (in bash) whether we are able to connect to the VNC display before actually starting the tests.

**Comment by devos50** [comment]
retest this please

**Pull request merged**